<launch>
<!--    <arg name="ns" default="camera_array"/>-->
<!--    <arg name="first_camera_name" default="cam0" />-->
<!--    <arg name="second_camera_name" default="cam1" />-->
<!--    <arg name="image_name" default="image_rect"/>-->

    <arg name="ns" default="kitti"/>
    <arg name="first_camera_name" default="camera_color_left" />
    <arg name="second_camera_name" default="camera_color_right" />
    <arg name="image_name" default="image_raw"/>

    <param name="/use_sim_time" value="true"/>

    <node pkg="viso2_ros" type="stereo_odometer" output="screen" name="viso2_ros">
        <param name="base_link_frame_id" value="base_footprint"/>
        <param name="ref_frame_change_method" value="1"/>
        <param name="approximate_sync" value="true"/>
        
<!--        <param name="max_features" value="50"/>-->
<!--        <param name="bucket_width" value="10"/>-->
<!--        <param name="bucket_height" value="10"/>-->

        <remap from="/stereo/left/image" to="/$(arg ns)/$(arg first_camera_name)/$(arg image_name)"/>
        <remap from="/stereo/right/image" to="/$(arg ns)/$(arg second_camera_name)/$(arg image_name)"/>
        <remap from="/stereo/left/camera_info" to="/$(arg ns)/$(arg first_camera_name)/camera_info"/>
        <remap from="/stereo/right/camera_info" to="/$(arg ns)/$(arg second_camera_name)/camera_info"/>
    </node>
</launch>